app/main.py

```diff
@@ -87,128 +1,31 @@
+from app.config.knights_config import KNIGHTS
+from app.knight_arena.prepare_for_battle import battle_preparations
+from app.peoples.knight_create import KnightCreate
+from app.knight_arena.fight import Fight
+from app.results.result import Result
 
 
-def battle(knightsConfig):
-    # BATTLE PREPARATIONS:
+def battle(knights_config: dict) -> dict:
+    lancelot = KnightCreate(knights_config["lancelot"])
+    battle_preparations(lancelot)
 
-    # lancelot
-    lancelot = knightsConfig["lancelot"]
+    arthur = KnightCreate(knights_config["arthur"])
+    battle_preparations(arthur)
 
-    # apply armour
-    lancelot["protection"] = 0
-    for a in lancelot["armour"]:
-        lancelot["protection"] += a["protection"]
+    mordred = KnightCreate(knights_config["mordred"])
```

Comment:
Again, consider refactoring the knight creation and preparation process to avoid repeating the same code block.